tools/RAiDER/delay.py

```diff
@@ -119,24 +114,20 @@ def tropo_delay(
     return wetDelay, hydroDelay
 
 
-def _get_delays_on_cube(dt, weather_model_file, wm_proj, ll_bounds, heights, los, crs, cube_spacing_m=None, look_dir='right', nproc=1):
+def _get_delays_on_cube(dt, weather_model_file, wm_proj, aoi, heights, los, crs, look_dir='right', nproc=1):
     """
     raider cube generation function.
     """
 
     # Determine the output grid extent here and clip output grid to multiples of spacing
 
-    snwe = transform_bbox(ll_bounds, src_crs=4326, dest_crs=crs)
-    out_spacing = get_output_spacing(cube_spacing_m, weather_model_file, wm_proj, crs)
-    out_snwe = clip_bbox(snwe, out_spacing)
-
+    out_snwe    = transform_bbox(aoi.bounds(), src_crs=4326, dest_crs=crs)
     logger.debug(f"Output SNWE: {out_snwe}")
-    logger.debug(f"Output cube spacing: {out_spacing}")
 
     # Build the output grid
     zpts = np.array(heights)
-    xpts = np.arange(out_snwe[2], out_snwe[3] + out_spacing, out_spacing)
-    ypts = np.arange(out_snwe[1], out_snwe[0] - out_spacing, -out_spacing)
+    xpts = np.arange(out_snwe[2], out_snwe[3] + aoi._output_spacing, aoi._output_spacing)
```

Comment:
It's not generally good practice to have `*._*` objects used by code outside the object. What I normally do is define a method e.g. "spacing()" that returns the underscored value.